### DISCUSSION TITLE
How can I create a meta tag template component?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#37730] [Help] [answered]
So I want to create a component for meta tags, that way I don't have to copy, and paste, and delete sections of meta tags for a new page.

I have some rough ideas doing it using props... but it isnt working fully...

Here is my sample:
from meta-data.js

```
import React from "react";

function Meta(props) {
  return (
    <div>
      <meta name="twitter:description" content="hello test" />
    </div>
  );
}

export default Meta;
```
^^  this is just a sample of one meta tag without including props in the tag.

then go to page:

```
import React from "react";
import Image from "next/image";
import Head from "next/head";
import Link from "next/link";
import Meta from "../../components/meta-data";
import Blogger from "../../components/blog-post";

const TestPost = () => {
  return (
    <div>
      <div>
        <Head>
          <Meta />
        </Head>
      </div>
      <Blogger
        title="Hello"
        body="this is the body"
        description="this is description"
      />
    </div>
  );
};

export default TestPost;
```

^^^  this is a page of a blog test trying to pull the meta tag and put it in the head of the page.....   when I go to the console in local host 3000  I can't see the meta tag.

grrr.. why wouldn't I be able to see the tag?....  actually, even the pages that have tags and I see them in the browser I can't find them in the localhost3000..

....  then the real deal ....  from here `content="hello test" `   I need to implement the props...

What could be a good way to write the props on this?  

Could be like this?

```
function Meta(props) {
  return (
    <div>
      <meta name="twitter:description" content={props.twitterdescription} />
    </div>
  );
}
```
And then at the page go like this? 

```
<div>
        <Head>
          <Meta twitterdescription=" This is a description"/>
        </Head>
      </div>
```

I can do a try and error and test it, but I can't see the tags on the localhost3000

**Comment by icyJoseph** [comment, marked as answer]
I am often the person that is always against installing new packages, but for this use case I'd highly recommend, https://www.npmjs.com/package/next-seo, which can handle all of these.
